### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by egichuhis**
Hello, please review this PR.

Required: Set up associations between classes and objects.

Changes:

Create the following classes and associations:

- class Classroom
- has-many/belongs-to relationship between Classroom and Student
- class Book
- class Rental
- many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental

In your IRB session, run the require_relative 'main' first before running the checks

`irb(main):001:0> require_relative 'main'`

oop

### CODE REVIEW

File: lib/classroom.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# classroom.rb
+class Classroom
+  attr_accessor :label, :students
```
Comment (DeliceLydia) [inline lib/classroom.rb]: - [x] I suggest that you make `students` an `attr_reader` we only need to read it

File: lib/student.rb [hunk 2]
```diff
@@ -1,9 +1,11 @@
-require_relative 'person'
-
+# student.rb
 class Student < Person
+  attr_accessor :classroom
```
Comment (DeliceLydia) [inline lib/student.rb]: - I suggest that you make. `classroom` an `attr_reader` as well as it's Read-only attribute in this case

File: lib/book.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# book.rb
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (DeliceLydia) [inline lib/book.rb:10]: - [x] Please create a  method for adding rentals and remember the book itself should be sent as a parameter to create the new rental achieving the `has-many` association
```ruby
 def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: lib/person.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
```
Comment (DeliceLydia) [inline lib/person.rb]: - [x] Please create a method for adding `rentals` in this class and remember the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
```ruby
 def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

File: lib/student.rb (hunk 2 above)
Comment (DeliceLydia) [reply lib/student.rb]: - [x] Please create a method that serves as the `belongs-to` side`(a student belongs to a classroom)`
```ruby
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @egichuhis, 

Good job so far!  There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Created required classes
- Followed git workflow
- Clean codebase


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: lib/person.rb (hunk 4 above)
Comment (egichuhis) [reply lib/person.rb]: Thanks @DeliceLydia for noting the above.

**Review by ghost — APPROVED** [review]
Hi @egichuhis , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Updated class associations  correctly
✅ Your documentation is polished and professional! 👍
✅ Went an extra mile for your `classroom` method in your `Student `class 👍 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**